src/Storages/StorageReplicatedMergeTree.h

```diff
@@ -103,6 +103,7 @@ class StorageReplicatedMergeTree final : public MergeTreeData
         const StorageID & table_id_,
         const String & relative_data_path_,
         const StorageInMemoryMetadata & metadata_,
+        ContextMutablePtr local_context_,
```

Comment:
(just some thoughts) I don't really like having `local_context` in this ctor because during server startup it's actually a copy of global context. It's not a problem right now because we only use it on CREATE, not on ATTACH. But later someone may accidentally use that context in a wrong way